stories/ui/Badge.js

```diff
@@ -18,10 +18,11 @@ const types = [
   'requires-payment',
 ];
 
-export type Props = {|
+export type Props = {
   props: BadgeProps,
-|};
+};
 
+// $FLOW_FIXME: I made PropsMatrixType exact but it's still not working here :/
```

Comment:
Flow wants all of the `Props` to follow the same shape. In other words, if all components can receive `wrapperClassNames` then all `Props` need to add that. Here is an example:

```diff
diff --git a/stories/ui/Badge.js b/stories/ui/Badge.js
index 81fbd05c0..ff6b2457a 100644
--- a/stories/ui/Badge.js
+++ b/stories/ui/Badge.js
@@ -18,11 +18,11 @@ const types = [
   'requires-payment',
 ];
 
-export type Props = {
+export type Props = {|
   props: BadgeProps,
-};
+  wrapperClassNames?: Array<string>,
+|};
 
-// $FLOW_FIXME: I made PropsMatrixType exact but it's still not working here :/
 function createPropsMatrix(chapter): Array<Props> {
   return [
     {
diff --git a/stories/ui/Rating.js b/stories/ui/Rating.js
index 173b444e8..93de9ada4 100644
--- a/stories/ui/Rating.js
+++ b/stories/ui/Rating.js
@@ -14,10 +14,10 @@ import 'amo/components/RatingManager/styles.scss';
 import 'amo/components/AddonReviewManagerRating/styles.scss';
 import 'amo/components/AddonReviewCard/styles.scss';
 
-export type Props = {
+export type Props = {|
   props: RatingProps,
   wrapperClassNames?: Array<string>,
-};
+|};
 
 function createPropsMatrix(): Array<Props> {
   return [
diff --git a/stories/utils.js b/stories/utils.js
index d2c511089..e43e8dfa8 100644
--- a/stories/utils.js
+++ b/stories/utils.js
@@ -2,10 +2,10 @@
 import * as React from 'react';
 import makeClassName from 'classnames';
 
-type PropsMatrixType = {
+type PropsMatrixType = {|
   props: Object,
   wrapperClassNames?: Array<string>,
-};
+|};
 
 type ChapterType = string | number | void;
 
```

Instead of copying/pasting `wrapperClassNames` to each one you can extend a shared type like `PropMatrixType`.

However, I think you could do this patch without `wrapperClassNames`, right? If so, I would suggest adding it later only when you need it, in a follow-up issue.

Reply:
well actually with a class from the styles imported (I believe from `AddonReviewManagerRating`), we see something like this:

![Alt text](https://monosnap.com/image/GKgwHaMZq0tztRuaEXRPQc7TrA1DZd.png)

This is seen when you're editing. If we leave this import out and do not add this class name then it would just look like the stars shown above. Is that okay? I figured we'd want to account for this..

Thanks for the Flow information. I thought if `PropsMatrixType` wasn't an exact type, it would allow for `Badge` and `Button` to have this. I guess it's moot tho if I am removing the `wrapperClassNames` above.

Reply:
I don't think we should add in simulations of each component that wraps `Rating` as part of this PR. If we do add them in a future PR, they should go in separate stories for each separate component. In other words, trying to manually set class names and include foreign styles will be hard to maintain; they could go out of sync with the real code very easily.

Reply:
okay, I'll remove all the external styles (including all container code to simulate the actual space there is).